### PR TITLE
feat: Add generators in syncing dimensions

### PR DESCRIPTION
**Pull request by Ashutosh619-sudo**
https://app.clickup.com/1864988/v/l/6-901603904304-1

## Summary by CodeRabbit

- **New Features**
	- Improved data synchronization methods for currencies, locations, classifications, departments, and subsidiaries, enhancing memory efficiency and performance.

- **Bug Fixes**
	- Resolved issues related to handling large datasets, ensuring smoother processing without memory overload.

- **Tests**
	- Updated test cases to align with the new generator-based data retrieval methods, improving test accuracy and reliability.

- **Documentation**
	- Adjusted data structure in test fixtures for locations to a nested list format, requiring updates in downstream data handling.

### CODE REVIEW

File: apps/netsuite/connector.py [hunk 1]
```diff
@@ -493,20 +493,20 @@ def sync_currencies(self):
...
-            currency_attributes.append(
-                {
-                    'attribute_type': 'CURRENCY',
-                    'display_name': 'Currency',
-                    'value': currency['symbol'],
-                    'destination_id': currency['internalId'],
-                    'active': True
-                }
-            )
+        for currency in currencies_generator:
```
Comment (ruuushhh) [inline apps/netsuite/connector.py:500]: Can you test this once?

I think this not how generator works
You will have to two for loops

Comment (ashwin1111) [reply apps/netsuite/connector.py:500]: +1 @Ashutosh619-sudo

File: apps/netsuite/connector.py [hunk 2]
```diff
@@ -493,20 +493,20 @@ def sync_currencies(self):
...
+                    {
+                        'attribute_type': 'CURRENCY',
+                        'display_name': 'Currency',
+                        'value': currency['symbol'],
+                        'destination_id': currency['internalId'],
+                        'active': True
+                    }
+                )
 
         DestinationAttribute.bulk_create_or_update_destination_attributes(
```
Comment (ashwin1111) [inline apps/netsuite/connector.py:511]: upserting to DB should also be per page

File: apps/netsuite/connector.py [hunk 3]
```diff
@@ -519,32 +519,33 @@ def sync_locations(self):
...
-                else:
-                    location_attributes.append({
-                        'attribute_type': 'LOCATION',
-                        'display_name': 'Location',
-                        'value': location['name'],
-                        'destination_id': location['internalId'],
-                        'active': not location['isInactive']
-                    })
 
         DestinationAttribute.bulk_create_or_update_destination_attributes(
```
Comment (ashwin1111) [inline apps/netsuite/connector.py:550]: upserting to DB should also be per page

File: apps/netsuite/connector.py [hunk 4]
```diff
@@ -578,19 +580,20 @@ def sync_departments(self):
...
+                if not department['isInactive']:
+                    department_attributes.append({
+                        'attribute_type': 'DEPARTMENT',
+                        'display_name': 'Department',
+                        'value': department['name'],
+                        'destination_id': department['internalId'],
+                        'active': not department['isInactive']
+                    })
 
         DestinationAttribute.bulk_create_or_update_destination_attributes(
```
Comment (ashwin1111) [inline apps/netsuite/connector.py:598]: take care everywhere

File: tests/test_netsuite/test_connector.py [hunk 5]
```diff
@@ -405,7 +405,7 @@ def test_sync_tax_items(mocker, db):
 
 def test_sync_currencies(mocker, db):
     mocker.patch(
-        'netsuitesdk.api.currencies.Currencies.get_all',
+        'netsuitesdk.api.currencies.Currencies.get_all_generator',
         return_value=data['get_all_currencies'][0]
```
Comment (ashwin1111) [inline tests/test_netsuite/test_connector.py:409]: response structure would change

File: tests/test_netsuite/test_connector.py [hunk 6]
```diff
@@ -325,7 +325,7 @@ def test_sync_subsidiaries(mocker, db):
 
 def test_sync_locations(mocker, db):
     mocker.patch(
-        'netsuitesdk.api.locations.Locations.get_all',
+        'netsuitesdk.api.locations.Locations.get_all_generator',
         return_value=data['get_all_locations']
```
Comment (ashwin1111) [inline tests/test_netsuite/test_connector.py:329]: response structure would change

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**